### ISSUE TITLE
Replace NULL with nullptr

### ISSUE THREAD

**Issue by davidebeatrici** [mumble-voip/mumble#3763]
From https://github.com/mumble-voip/mumble/pull/3756#discussion_r317316435:

> We always used `NULL`, for consistency.
>
> The issue is that very recent compilers print a warning for each use of `NULL` instead of `nullptr`.
>
> `nullptr` was introduced in C++11, which is our minimum requirement starting with 1.4; Qt is replacing `Q_NULLPTR` (macro defined to either `nullptr` or `NULL`, depending on the compiler) with `nullptr` everywhere: https://bugreports.qt.io/browse/QTBUG-52736

**Comment by Kissaki** [comment]
> Qt is replacing `Q_NULLPTR` with `nullptr` everywhere

They are replacing it for Qt 5.7/from Qt 5.7 onward

> All references to Q_NULLPTR must be replaced with nullptr for Qt 5.7.0 release.

**Comment by Kissaki** [comment]
> C++11, which is our minimum requirement starting with 1.4

Who defined that/Where is that defined? What’s Qts or other projects state on C++11 requirement? Is compiler coverage on old Linux distro version supportive enough?

**Comment by davidebeatrici** [comment]
> They are replacing it for Qt 5.7/from Qt 5.7 onward

Yes, what I meant to say is that we shouldn't replace `NULL` with `Q_NULLPTR`.

> Who defined that/Where is that defined?

We discussed about it months ago on IRC, when I attempted to update our current build environment's dependencies.

> What’s Qts or other projects state on C++11 requirement?

Qt 5.7+ requires C++11 (as mentioned in **QTBUG-52736**).

> Is compiler coverage on old Linux distro version supportive enough?

As far as I know, there are no distributions (relatively recent, e.g. Ubuntu >= 14.04) shipping compilers without C++11 support.
